eng/mgmt/changelog/src/main/java/com/azure/resourcemanager/tools/changelog/changelog/DefinitionStageChangeLog.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.tools.changelog.changelog;
+
+import com.azure.resourcemanager.tools.changelog.utils.AllMethods;
+import com.azure.resourcemanager.tools.changelog.utils.ClassName;
+import com.azure.resourcemanager.tools.changelog.utils.MethodName;
+import japicmp.model.JApiMethod;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class DefinitionStageChangeLog extends ChangeLog {
+    private String parentClass;
+    private List<Set<JApiMethod>> oldMethodStages;
+    private List<Set<JApiMethod>> newMethodStages;
+
+    DefinitionStageChangeLog(Map<String, AllMethods> allStages, String parentClass) {
+        this.parentClass = parentClass;
+        oldMethodStages = new ArrayList<>();
+        newMethodStages = new ArrayList<>();
+        AllMethods blankStage = allStages.entrySet().stream().filter(x -> ClassName.name(x.getKey()).equals("Blank")).findAny().get().getValue();
+        calcMethodStages(blankStage, allStages, oldMethodStages, x -> x.getReturnType().getOldReturnType());
+        calcMethodStages(blankStage, allStages, newMethodStages, x -> x.getReturnType().getNewReturnType());
+        calcChangeLog();
+    }
+
+    void calcMethodStages(AllMethods blankStage, Map<String, AllMethods> allStages, List<Set<JApiMethod>> results, Function<JApiMethod, String> getReturnType) {
+        results.add(new HashSet<>(blankStage.getMethods()));
+        Set<JApiMethod> used = new HashSet<>(blankStage.getMethods());
+        for (int i = 0; i < results.size(); ++i) {
+            Set<JApiMethod> nextStage = new HashSet<>();
+            results.get(i).forEach(method -> {
+                String returnType = getReturnType.apply(method);
+                if (allStages.containsKey(returnType)) {
+                    allStages.get(returnType).getMethods().forEach(nextMethod -> {
+                        if (!used.contains(nextMethod)) {
+                            nextStage.add(nextMethod);
+                            used.add(nextMethod);
+                        }
+                    });
+                }
```

Comment:
More comments please. Hard to understand by code alone.

Reply:
@xccc-msft Can you understand it?

Reply:
Suggestion:

allStages -> stageToMethodsMapping
results -> methodsByStageIndex
nextStage -> methodsInNextStage
returnType -> nextStageName

Reply:
done